File: Third task/thirdTask.exs
defmodule Main do
  def filter(num) do
    sum = num
          |> String.graphemes()        # Разбиваем строку на отдельные символы
          |> Enum.map(&String.to_integer/1)  # Преобразуем каждый символ в число
          |> Enum.sum()                # Суммируем все цифры
    
    sum > 10  # Возвращаем true если сумма > 10
  end



  def main do
    # Заданный массив чисел 
    numbers = ["124", "9829", "9872", "2", "456"]
    
    # Счётчик подходящих чисел
    counter = Enum.count(numbers, &filter/1)
    
    # Вывод результатов
    IO.puts("Массив: #{Enum.join(numbers, ", ")}")
    IO.puts("Результат: #{counter}")  
  end
end

Main.main()

File: Third task/thirdTask.cpp
#include <iostream>
using namespace std;


bool filter(string num){
    int sum = 0;
    for(char k : num){
        k = k - '0';  // Преобразование символа '5' в число 5
        sum += k;   
    }
    return (sum > 10) ? true : false;  // Тернарный оператор
}

int main(){
    cout << "Введитe количество чисел" << endl;
    int n, counter = 0;  // n сколько чисел; counter счётчик подходящих
    cin >> n;
    
    for (int i = 0; i < n; i++){
        cout << "Введите число: " << endl;
        string num;
        cin >> num;

        if (filter(num)){  // Если сумма цифр > 10
            counter++;     // Увеличиваем счётчик
        }
    }
    cout << "Результат: " << counter << endl;
    return 0;
}
